File: chrome-extension/icons/generate_icons.py
#!/usr/bin/env python3
"""
Simple icon generator for Voyce Chrome Extension
Creates placeholder icons with gradient background and microphone symbol
Requires: PIL/Pillow (pip install Pillow)
"""

try:
    from PIL import Image, ImageDraw, ImageFont
except ImportError:
    print("Pillow not installed. Install with: pip install Pillow")
    exit(1)

def create_icon(size):
    """Create a gradient icon with white circle"""
    # Create image
    img = Image.new('RGB', (size, size))
    draw = ImageDraw.Draw(img)

    # Draw gradient background (purple to blue)
    for y in range(size):
        # Interpolate between colors
        r = int(102 + (118 - 102) * y / size)
        g = int(126 + (75 - 126) * y / size)
        b = int(234 + (162 - 234) * y / size)
        draw.line([(0, y), (size, y)], fill=(r, g, b))

    # Draw white circle in center
    center = size // 2
    radius = size // 3
    draw.ellipse(
        [center - radius, center - radius, center + radius, center + radius],
        fill='white'
    )

    # Draw microphone shape (simplified)
    mic_height = size // 4
    mic_width = size // 6
    mic_x = center - mic_width // 2
    mic_y = center - mic_height // 2

    # Mic body (rounded rectangle approximation)
    draw.ellipse(
        [mic_x, mic_y, mic_x + mic_width, mic_y + mic_height],
        fill=(102, 126, 234)
    )

    # Save
    filename = f'icon{size}.png'
    img.save(filename, 'PNG')
    print(f'Created {filename}')

# Generate all three sizes
for size in [16, 48, 128]:
    create_icon(size)

print('\nIcons created successfully!')
print('You can now load the extension in Chrome.')

File: chrome-extension/icons/generate_icons.js
#!/usr/bin/env node
/**
 * Simple icon generator using Canvas
 * Run with: node generate_icons.js
 * Requires: npm install canvas
 */

const fs = require('fs');

try {
  const { createCanvas } = require('canvas');

  function createIcon(size) {
    const canvas = createCanvas(size, size);
    const ctx = canvas.getContext('2d');

    // Create gradient background
    const gradient = ctx.createLinearGradient(0, 0, size, size);
    gradient.addColorStop(0, '#667eea');
    gradient.addColorStop(1, '#764ba2');
    ctx.fillStyle = gradient;
    ctx.fillRect(0, 0, size, size);

    // Draw white circle (microphone base)
    ctx.fillStyle = 'white';
    ctx.beginPath();
    ctx.arc(size / 2, size / 2, size / 3, 0, Math.PI * 2);
    ctx.fill();

    // Draw simple microphone icon
    ctx.fillStyle = '#667eea';
    const micWidth = size / 6;
    const micHeight = size / 4;
    const x = (size - micWidth) / 2;
    const y = (size - micHeight) / 2;

    // Mic capsule
    ctx.beginPath();
    ctx.roundRect(x, y, micWidth, micHeight, micWidth / 2);
    ctx.fill();

    // Save to file
    const buffer = canvas.toBuffer('image/png');
    const filename = `icon${size}.png`;
    fs.writeFileSync(filename, buffer);
    console.log(`Created ${filename}`);
  }

  // Generate all sizes
  [16, 48, 128].forEach(createIcon);
  console.log('\nIcons created successfully!');

} catch (error) {
  console.error('Canvas module not installed.');
  console.error('Install with: npm install canvas');
  console.error('\nAlternatively, use the Python script or create icons manually.');
  console.error('See icons/README.md for details.');
  process.exit(1);
}
